### PR TITLE
Fix functionality to allow a value of 0 to be configured for the appropriate integer attributes and refactor STP module code

### PR DESCRIPTION
**Pull request by stalabi1**
##### SUMMARY

- I fixed the functionality of the code to allow a value of 0 to be configured for the appropriate integer attributes.
- I removed all of the default definitions from the argspec because the definitions were making some configuration cases impossible to implement, e.g. the deletion of an STP interface by interface name that has all defaults configured.
- I refactored the code to reduce the number of nested loops in the case of searching for instance matches. I condensed a few of the get delete request methods.
- I added more UT test cases to increase code coverage.
- I updated the facts code to properly handle empties allowing me to remove a few 'remove_empties' calls from config.

##### GitHub Issues

| GitHub Issue #472  |
| -------------- |

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

sonic_stp

##### OUTPUT

[regression-2024-11-11-14-59-56.html.pdf](https://github.com/user-attachments/files/17708910/regression-2024-11-11-14-59-56.html.pdf)
[merge_mst_id_0_output.log](https://github.com/user-attachments/files/17562751/merge_stp_output.log)

##### ADDITIONAL INFORMATION

Current SONiC behavior doesn't allow me to to configure any other mst-id when mst-id 0 is configured.
##### Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility or have provided any relevant "breaking_changes" descriptions in a "fragment" file in the "changelogs/fragments" directory of this repository.
- [x] I have provided a summary for this PR in valid "fragment" file format in the "changelogs/fragments" directory of this repository branch. Reference : [Ansible Change Log Document](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)

### CODE REVIEW

**Comment by TafkaMax** [comment]
Tested this branch and it fixes the issue I had.

**Review by kerry-meyer — CHANGES_REQUESTED** [review]
This change set, the additional test cases, and the corresponding test results looks good.

Thanks for adding efficiency improvements and reductions in the existing code along with the basic fix.

I am posting several minor comments/questions and change requests, but am not requesting any significant algorithmic changes.

File: plugins/module_utils/network/sonic/config/stp/stp.py [hunk 1]
```diff
@@ -1221,172 +1206,171 @@ def get_delete_rapid_pvst_vlan_cfg_attr(self, vlan_id, attr):
...
+            portfast = stp_global.get('portfast')
+            bridge_priority = stp_global.get('bridge_priority')
 
             if loop_guard is False:
                 stp_global.pop('loop_guard')
             if bpdu_filter is False:
                 stp_global.pop('bpdu_filter')
             if portfast is False:
                 stp_global.pop('portfast')
-            if hello_time == 2:
```
[the remark this answers is not in the document]
Comment (kerry-meyer) [reply plugins/module_utils/network/sonic/config/stp/stp.py:1255]: Thanks for explaining.

Resolved.

**Review by kerry-meyer — APPROVED** [review]
All current proposed changes and corresponding test results look good.

Thank you for not only fixing the problem described by the customer, but for taking the initiative to fix all related exposures in the STP resource module.

Approved.

**Pull request merged**